db_management.py

```diff
@@ -1,31 +1,48 @@
-import sqlite3
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+import configurations
+from signup_utils import SignupResults
+from models.User import User
 
+engine = create_engine(configurations.connection_string, connect_args={'check_same_thread': False})
+Base = declarative_base()
 
-def new_user_creation(username_text, pass_text):
-    conn = sqlite3.connect('data_base.db')
-    curs = conn.cursor()
-    params = (username_text, pass_text)
-    curs.execute("INSERT INTO users VALUES (NULL, ?, ?)", params)
-    conn.commit()
-    conn.close()
-    return True
 
+def create_connection():
+    return engine.connect()
 
-def is_username_taken(str_input):
-    conn = sqlite3.connect('data_base.db')
-    curs = conn.cursor()
-    curs.execute('SELECT * FROM users WHERE Username = ?', (str_input,)).fetchall()
-    if len(curs.fetchall()) == 0:
-        return True
-    return False
 
+def create_tables():
+    Base.metadata.create_all(engine)
 
-def does_user_exist(str_username, str_pass):
-    conn = sqlite3.connect('data_base.db')
-    curs = conn.cursor()
-    curs.execute('''SELECT * FROM users WHERE Username = ? AND Password = ?;''',
-                 (str_username, str_pass))
-    if len(curs.fetchall()) == 0:
-        return False
-    return True
 
+def check_login_parameters(username: str, password: str) -> bool:
+    session_made = sessionmaker(bind=engine)
+    session = session_made()
+
+    user_count = session.query(User).filter(
+        User.Username == username and User.Password == password).count()
+    return user_count > 0
```

Comment:
I think we should change this to:
`return user_count == 1`
What happens if you have 2 rows for this username / password (maybe because of some manual operation)? You want to know if there's exactly 1 match here, right?